packages/office-ui-fabric-react/src/components/pickers/BasePicker.tsx

```diff
@@ -746,12 +743,17 @@ export class BasePicker<T, P extends IBasePickerProps<T>> extends BaseComponent<
     );
   };
 
-  protected addItemByIndex = (index: number): void => {
-    this.addItem(this.suggestionStore.getSuggestionAtIndex(index).item);
+  protected completeSelection = (item: T) => {
+    this.addItem(item);
```

Comment:
Just checking my understanding - this addition of checking current input prevents it from clearing selection if it's already been selected? If so, shouldn't the updateValue call with empty string happen after the if?

Reply:
Do you mean `if(this.input.current)` ? That's just checking to make sure that the input.current already exists before clearing it out. That's a separate thing from clearing the currently entered value. Both things should happen but are independent of one another. `updateValue` triggers the clearing of the suggested items that are available to the user. That should always get cleared out/get called. 

It's possible that the `updateValue` call should be contained within the if statement. But I'm not sure it needs to be.